modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/CommandProcessor.java

```diff
@@ -1204,6 +1204,29 @@ else if (createTbl.wrapKey()) {
         return res;
     }
 
+    private static int convertH2ColumnPrecision(TypeInfo type) {
+        if (type.getValueType() == Value.DECIMAL) {
+            if (type.getPrecision() < H2Utils.DECIMAL_DEFAULT_PRECISION)
+                return (int)type.getPrecision();
+        }
+        else if (type.getValueType() == Value.STRING ||
+            type.getValueType() == Value.STRING_FIXED ||
+            type.getValueType() == Value.STRING_IGNORECASE) {
+            if (type.getPrecision() < H2Utils.STRING_DEFAULT_PRECISION)
+                return (int)type.getPrecision();
+        }
+
+        return -1;
```

Comment:
Let's avoid magic numbers and add add a constants `UNDEFINED_PRECISION` & `UNDEFINED_SCALE`

Reply:
Fixed, thanks.